README.md

```diff
@@ -21,56 +21,79 @@
 ```
 **Where:**
-   - *secret* is a String to be used for session encoding. It should be a large unguessable string, say minimum 14 characters long.
-   - *cookie* is a list of options for session's cookies. The options are (specified in `CookieParameter` enumeration): `name` - cookie's name, defaults to "kitura-session-id", `path` - cookie's Path attribute defaults to "/", `secure` - cookie's Secure attribute, false by default, and `maxAge` - an NSTimeInterval with cookie's expiration time in seconds, defaults to -1.0, i.e., no expiration.
-   - *store* is an instance of a plugin for session backing store that implements `Store` protocol. If not set, `InMemoryStore` is used.
+   - *secret* is a String to be used for session encoding. It should be a large unguessable string, a minimum of 14 characters long.
+   - *cookie* is a list of options for session's cookies. The options are (as specified in the `CookieParameter` enumeration):
+     - `name` - cookie's name, defaults to "kitura-session-id".
+     - `path` - cookie's path attribute, this defines the path for which this cookie should be supplied. Defaults to "/" which means allow any path.
+     - `secure` - cookie's secure attribute, this indicates whwether the cookie should be provided only over secure (https) connections. Defaults to false.
+     - `maxAge` - cookie's maxAge attribute, that is, the maximum age (in seconds) from the time of issue that the cookie should be kept for. Defaults to -1.0, i.e. no expiration.
+   - *store* is an instance of a plugin for a session backing store that implements the `Store` protocol. If not set, `InMemoryStore` is used.
    <br>
 
-   The last two parameters are optional.
+   The *cookie* and *store* parameters are optional.
 
-   <br>
-   The *secret* parameter is used to secure the session ID and ensure that the session ID cannot be guessed. *Secret* is used to derive a pair of encryption and signature keys via PBKDF2 and a fixed IV to make the session ID cookie be authenticated encrypted. *Secret* isn't used directly to encrypt or compute the MAC of the cookie.
+   The *secret* parameter is used to secure the session ID and ensure that the session ID cannot be guessed. *Secret* is used to derive a pair of encryption and signature keys via PBKDF2 and a fixed IV to make the session ID cookie be authenticated and encrypted. *Secret* isn't used directly to encrypt or compute the MAC of the cookie.
 
 ## Example
 
-This is an example of `Session` middleware with [`KituraSessionRedis`](https://github.com/IBM-Swift/Kitura-Session-Redis) plugin:
+This is an example using `Session` middleware with the [`KituraSessionRedis`](https://github.com/IBM-Swift/Kitura-Session-Redis) plugin:
 
 ```swift
+import Kitura
 import KituraSession
 import KituraSessionRedis
 
 let redisStore = RedisStore(redisHost: host, redisPort: port)
 let session = Session(secret: "Some secret", store: redisStore)
 router.all(middleware: session)
 ```
```

Comment:
Could we add using the session within a route to this example. E.g:
 router.post("/session") {request, response, next in
            request.session?["key"] = "value"
            next()
        }
router.get("/session") {request, response, next in
            let value = request.session?["key"] as? String
            next()
        }